File: template.cpp
#include <iostream>
#include <string>

std::string MacCompony(const int macDigit){
	switch (macDigit){
		// ***insert_point***
		case 123: return "good";
		// ***insert_point***
		default: return "unknown";
	}
	return "unknown";
}

int main(int argc, char* argv[]) {
	for (int i = 1; i < argc; ++i) {
		const std::string mac = std::string(argv[i]);
		int macDigit = 0;
		int addTimes = 0;
		for(char c: mac){
			if (isalnum(c)) {
				c = toupper(c);
				if (c >= 'A' && c <= 'F') {
					macDigit = macDigit * 16 + 10 + c - 'A';
				}
				else {
					macDigit = macDigit * 16 + c - '0';
				}
				addTimes++;
				if (addTimes >= 6) {
					break;
				}
			}
		}
		const std::string ans = MacCompony(macDigit);
		std::cout << mac << " " << ans << std::endl;
	}
	return 0;
}

File: template.rs
use std::env;
use phf::phf_map;


static OUI_DATA: phf::Map<&'static str, &'static str> = phf_map! {
    // ***insert_point***
    "286FB9" => "Nokia Shanghai Bell Co., Ltd.",
    // ***insert_point***
};


fn main() {
    let args: Vec<String> = env::args().collect();
    for mac in args.iter().skip(1) {
        let mut mac_alpha = String::new();
        let mut add_times: u32 = 0;
        for c in mac.chars() {
            if c.is_alphanumeric() {
                mac_alpha.push(c.to_ascii_uppercase());
                add_times += 1;
                if add_times >= 6 {
                    break;
                }
            }
        }
        if OUI_DATA.contains_key(&mac_alpha) {
            println!("{} {}", mac, OUI_DATA.get(&mac_alpha).unwrap());
        } else {
            println!("{} unknown", mac);
        }
    }
}
